### PR TITLE
ALCA and radar_interface work

### PR DESCRIPTION
**Pull request by BogGyver**
- Improve ALCA integration by moving all signal publishers and changes into the ALCA_module
- increase # of errors before generating CAN error from radar to 3
- improve logic for parsing a full block of values by checking for start and end message presence
- re-add logic for angle_bias (but keep disabled)
- check at top if Index and Index2 from the 2 messages for each object match

### CODE REVIEW

File: selfdrive/controls/lib/pathplanner.py [hunk 1]
```diff
@@ -24,13 +23,12 @@ def calc_states_after_delay(states, v_ego, steer_angle, curvature_factor, steer_
 
 class PathPlanner(object):
   def __init__(self, CP):
-    self.LP = LanePlanner()
+    self.LP = LanePlanner(True)
```
Comment (rafcabezas) [inline selfdrive/controls/lib/pathplanner.py]: At the point of use, here, it's not clear what True/False represent, consider specifying the argument name.
```python
self.LP = LanePlanner(withAlca=True)
```

File: selfdrive/controls/lib/lane_planner.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def calc_d_poly(l_poly, r_poly, p_poly, l_prob, r_prob, lane_width):
 
 
 class LanePlanner(object):
-  def __init__(self):
+  def __init__(self,withAlca):
```
Comment (rafcabezas) [inline selfdrive/controls/lib/lane_planner.py]: Consider prepending the parameter such that it's clear it's a boolean. Example `isAlcaEnabled` `shouldUseAlca`

File: selfdrive/controls/radard.py [hunk 3]
```diff
@@ -385,7 +379,7 @@ def radard_thread(gctx=None):
 
   rk = Ratekeeper(rate, print_delay_threshold=None)
   RD = RadarD(mocked, RI)
-  MP = LanePlanner()
+  MP = LanePlanner(False)
```
Comment (rafcabezas) [inline selfdrive/controls/radard.py]: At the point of use, here, it's not clear what True/False represent, consider specifying the argument name.
```python
self.LP = LanePlanner(withAlca=False)
```

File: selfdrive/car/tesla/radar_interface.py [hunk 4]
```diff
@@ -15,6 +15,8 @@
 RADAR_B_MSGS = list(range(0x311, 0x36F, 3))
 OBJECT_MIN_PROBABILITY = 20.
 CLASS_MIN_PROBABILITY = 20.
+RADAR_MESSAGE_FREQUENCY = 0.050 * 1e9 #time in ns, radar sends data at 0.06 s
+VALID_THRESHOLD = 10
```
Comment (rafcabezas) [inline selfdrive/car/tesla/radar_interface.py]: Consider specifying what this threshold represents.
Ex: `VALID_MESSAGE_COUNT_THRESHOLD`

File: selfdrive/controls/lib/lane_planner.py (hunk 2 above)
Comment (BogGyver) [reply selfdrive/controls/lib/lane_planner.py]: done

File: selfdrive/controls/radard.py (hunk 3 above)
Comment (BogGyver) [reply selfdrive/controls/radard.py]: done

File: selfdrive/car/tesla/radar_interface.py (hunk 4 above)
Comment (BogGyver) [reply selfdrive/car/tesla/radar_interface.py]: done

**Review by rafcabezas — APPROVED** [review]

**Pull request merged**